profiles-jar-examples/main-app/src/main/java/com/c12e/cortex/examples/StreamingDataSource.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2022 Cognitive Scale, Inc. All Rights Reserved.
+ *
+ *  See LICENSE.txt for details.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.c12e.cortex.examples;
+
+import com.c12e.cortex.phoenix.DataSource;
+import com.c12e.cortex.profiles.CortexSession;
+import com.c12e.cortex.profiles.client.LocalSecretClient;
+import com.c12e.cortex.profiles.module.CortexDataSourcePair;
+import io.delta.tables.DeltaTable;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SaveMode;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.streaming.StreamingQueryListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+import java.util.HashMap;
+import java.util.concurrent.TimeoutException;
+
+@CommandLine.Command(name = "ds-streaming", description = "Example Streaming DataSource", mixinStandardHelpOptions = true)
+public class StreamingDataSource extends  BaseCommand implements Runnable {
+    @CommandLine.Option(names = {"-p", "--project"}, description = "Project to use", required = true)
+    String project;
+
+    @CommandLine.Option(names = {"-d", "--data-source"}, description = "DataSource Name", required = true)
+    String dataSourceName;
+
+    @CommandLine.Spec
+    private CommandLine.Model.CommandSpec cmdSpec;
+
+    private static final String AWS_CONNECTION_SECRET_ENV = "CONN_AWS_SECRET";
+
+    private static void checkRequiredSecrets() {
+        if (System.getenv(AWS_CONNECTION_SECRET_ENV) == null) {
+            System.err.println(String.format("Missing environment variable '%s' for local secrets client", AWS_CONNECTION_SECRET_ENV));
+            System.exit(2);
+        }
+    }
+
+    Logger logger = LoggerFactory.getLogger(StreamingDataSource.class);
+
+    /**
+     * Code to run for a command
+     */
+    @Override
+    public void run() {
+        //get spark session
+        SparkSession session = getSparkSession(getDefaultProps());
+
+        //create local secrets map for use in non-cluster env
+        checkRequiredSecrets();
+        LocalSecretClient.LocalSecrets localSecrets = new LocalSecretClient.LocalSecrets();
+        localSecrets.setSecretsForProject(project, new HashMap() {{
+                    put("aws-secret", System.getenv(AWS_CONNECTION_SECRET_ENV));
```

Comment:
Note for future - might want to make the secret name a parameter. (Or leave as a default to force the user to update the code to match their connection)